src/Plugin.Permissions.Tizen/PermissionsImplementation.cs

```diff
@@ -0,0 +1,184 @@
+					}
+					catch
+					{
+						return Task.FromResult(PermissionStatus.Unknown);
+					}
+				case Permission.Location:
+					try
+					{
+						if (Privilege.GetPrivacyPrivilegeStatus(privilegeLocation))
+							return Task.FromResult(PermissionStatus.Granted);
+						else
+							return Task.FromResult(PermissionStatus.Denied);
+					}
+					catch
+					{
+						return Task.FromResult(PermissionStatus.Unknown);
+					}
+				case Permission.Microphone:
+					try
+					{
+						if (Privilege.GetPrivacyPrivilegeStatus(privilegeRecorder))
+							return Task.FromResult(PermissionStatus.Granted);
+						else
+							return Task.FromResult(PermissionStatus.Denied);
+					}
+					catch
+					{
+						return Task.FromResult(PermissionStatus.Unknown);
+					}
+				case Permission.Phone:
+					try
+					{
+						if (Privilege.GetPrivacyPrivilegeStatus(privilegeCall))
+							return Task.FromResult(PermissionStatus.Granted);
+						else
+							return Task.FromResult(PermissionStatus.Denied);
+					}
+					catch
+					{
+						return Task.FromResult(PermissionStatus.Unknown);
+					}
+				case Permission.Photos:
+					return Task.FromResult(PermissionStatus.Unknown);
+				case Permission.Reminders:
+					return Task.FromResult(PermissionStatus.Unknown);
+				case Permission.Sensors:
+					try
+					{
+						if (Privilege.GetPrivacyPrivilegeStatus(privilegeHealthInfo))
+							return Task.FromResult(PermissionStatus.Granted);
+						else
+							return Task.FromResult(PermissionStatus.Denied);
+					}
+					catch
+					{
+						return Task.FromResult(PermissionStatus.Unknown);
+					}
+				case Permission.Sms:
+					try
+					{
+						if (Privilege.GetPrivacyPrivilegeStatus(privilegeMessageRead) &&
+							Privilege.GetPrivacyPrivilegeStatus(privilegeMessageWrite))
+							return Task.FromResult(PermissionStatus.Granted);
+						else
+							return Task.FromResult(PermissionStatus.Denied);
+					}
+					catch
+					{
+						return Task.FromResult(PermissionStatus.Unknown);
+					}
+				case Permission.Storage:
+					try
+					{
+						if (Privilege.GetPrivacyPrivilegeStatus(privilegeExternalStorage))
+							return Task.FromResult(PermissionStatus.Granted);
+						else
+							return Task.FromResult(PermissionStatus.Denied);
+					}
+					catch
+					{
+						return Task.FromResult(PermissionStatus.Unknown);
+					}
+				case Permission.Speech:
+					return Task.FromResult(PermissionStatus.Unknown);
+				default:
+					return Task.FromResult(PermissionStatus.Unknown);
+			}
+		}
+
+		/// <summary>
+		/// Requests the permissions from the users
+		/// </summary>
+		/// <returns>The permissions and their status.</returns>
+		/// <param name="permissions">Permissions to request.</param>
+		public Task<Dictionary<Permission, PermissionStatus>> RequestPermissionsAsync(params Permission[] permissions)
+		{
+			var results = new Dictionary<Permission, PermissionStatus>();
+
+			foreach (var permission in permissions)
```

Comment:
Is there no way to request a permission? I would think it would return the actual value of the permission and not Unknown.

Reply:
@JongHeonChoi could you reply to James' question? It will help us a lot to merge these changes